src/lib/ability.test.ts

```diff
@@ -0,0 +1,86 @@
+import { test, expect } from 'vitest'
+
+import type { Session, UserOrgRoles } from './types'
+import { faker } from '@faker-js/faker'
+import { subject } from '@casl/ability'
+import { defineAbilityFor } from './ability'
+import { StudyStatus } from '@/database/types'
+
+const createAbilty = (roles: Partial<UserOrgRoles> = {}) => {
+    const session: Session = {
+        user: {
+            id: faker.string.uuid(),
+            email: faker.internet.email(),
+        },
+        team: {
+            id: faker.string.uuid(),
+            name: faker.company.name(),
+        },
+        roles: {
+            isAdmin: false,
+            isResearcher: false,
+            isReviewer: false,
+            ...(roles || {}),
+        },
+    }
+    return { ability: defineAbilityFor(session), session }
+}
+
+test('researcher role', () => {
+    const { ability, session } = createAbilty({ isReviewer: true })
```

Comment:
test is incorrectly configured to test reviewer's permissions for researcher role. Probably needs other test case name.